app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -10,6 +10,21 @@ class MetaRPCClient {
     this.requests = new Map();
     this.connectionStream.on('data', this.handleResponse.bind(this));
     this.connectionStream.on('end', this.close.bind(this));
+    this.responseHandled = false;
```

Comment:
This client is written to handle multiple requests simultaneously. That's why the request callbacks are tracked with a map, with ids as the key. This looks like it assumes there is only one request happening at a time. As such this seems to be totally broken.